### PR TITLE
10295: Remove twisted.web.client.getPage and friends

### PR DESCRIPTION
**Pull request by twm**
GHSA-92x2-jw7w-xvvx

## Scope and purpose

Add a few words about why this PR is needed and what is its scope.

Add any comments about trade-offs (if any) made in this PR and the reasoning behind them.

Add mentions of things that are not covered here and are planed to be done in separate PRs.

## Contributor Checklist:

* [x] The associated ticket in Trac is here: https://twistedmatrix.com/trac/ticket/10295
* [x] I ran `tox -e lint` to format my patch to meet the [Twisted Coding Standard](https://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html)
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: [News files](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles))
* [x] The title of the PR starts with the associated Trac ticket number (without the `#` character).
* [x] I have updated the automated tests and checked that all checks for the PR are green.
* [x] I have submitted the associated Trac ticket for review by adding the word `review` to the keywords field in Trac, and putting a link to this PR in the comment; it shows up in https://twisted.reviews/ now.

### CODE REVIEW

**Review by glyph — APPROVED** [review]
*incredible*

image


🚢

File: src/twisted/web/newsfragments/10295.removal [hunk 1]
```diff
@@ -0,0 +1 @@
+twisted.web.client.getPage, twisted.web.client.downladPage, and the associated implementation classes (HTTPPageGetter, HTTPPageDownloader, HTTPClientFactory, HTTPDownloader) have been removed because they do not segregate cookies by domain. They were deprecated in Twisted 16.7.0 in favor of twisted.web.client.Agent. GHSA-92x2-jw7w-xvvx.
```
Comment (DMRobertson) [inline src/twisted/web/newsfragments/10295.removal:1]: Typo here, I think: downladPage -> downloadPage

**Comment by andrewshadura** [comment]
> Add a few words about why this PR is needed and what is its scope.
> 
> Add any comments about trade-offs (if any) made in this PR and the reasoning behind them.
> 
> Add mentions of things that are not covered here and are planed to be done in separate PRs.

> mumble mumble security

Seriously, *this* gets merged without a word? Wow.

**Comment by twm** [comment]
Hi @andrewshadura, the [security policy](https://twistedmatrix.com/trac/wiki/Security) says to create a placeholder ticket without any information. I will fill out that ticket once the advisory is made public upon release of Twisted 22.1.0. The changelog entry does say most of what needs be said, though &mdash; these APIs have been obsolete for a decade and deprecated for years, and they double the work of addressing every security issue reported against twisted.web.client.

**Comment by twm** [comment]
I have updated https://twistedmatrix.com/trac/ticket/10295

**Pull request merged**